lib/puppet/http/resolver/server_list.rb

```diff
@@ -63,7 +63,12 @@ def resolve(session, name, ssl_context: nil, canceled_handler: nil)
         Puppet.log_exception(detail, _("Puppet server %{host}:%{port} is unavailable: %{code} %{reason}") %
                              { host: service.url.host, port: service.url.port, code: detail.response.code, reason: detail.response.reason })
       rescue Puppet::HTTP::HTTPError => detail
-        Puppet.log_exception(detail, _("Unable to connect to server from server_list setting: %{detail}") % {detail: detail})
+        if server.equal?(@server_list_setting.value.last)
```

Comment:
It's possible, though unlikely that a server might appear multiple times in list, in which case we would call `log_exception` multiple times:

```
server_list: a, b, a
```
Might be clearer to pass the index when iterating so we know when we've reached the end?

```ruby
@server_list_setting.value.each_with_index do |server, i|
  ...
  rescue Puppet::HTTP::HTTPError => detail
    if i < @server_list_setting.value.length - 1
     # try next server case
   else
    # reached end of list case
  end
end
```